Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -117,12 +117,14 @@ public IEnumerable<TJob> ListJobs(bool forceRefreshCache)
         {
             lock (jobsListCacheLockObj)
             {
-                if (JobListCache == null || forceRefreshCache)
+                var jobList = JobListCache;
+                if (jobList == null || forceRefreshCache)
                 {
-                    JobListCache = ListJobsInternal();
+                    jobList = ListJobsInternal();
+                    JobListCache = jobList;
                 }
+                return jobList;
             }
-            return JobListCache;
```

Comment:
the fix here is we return JobListCache outside the lock.   This could be null (I ran into nullref myself) as it is being set to null by other code path.   Use local variable to make it more robust.  In fact, we could remove the `lock(jobsListCacheLockObj)`.